### PR TITLE
Prevent writing unchanged files when fixing

### PR DESCRIPTION
**Pull request by rvanvelzen**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Hi ! Thanks for the PR.
Just to know ; is there a performance-improvement purpose or is there another reason to this change ? (Which seems logic)

File: src/Runner/Linter.php [hunk 1]
```diff
@@ -77,8 +77,11 @@ public function run(iterable $files, Ruleset $ruleset, ?FixerInterface $fixer =
 
             if (null !== $fixer) {
                 try {
-                    $content = $fixer->fixFile($content, $ruleset);
-                    file_put_contents($filePath, $content);
+                    $newContent = $fixer->fixFile($content, $ruleset);
+                    if ($newContent !== $content) {
+                        file_put_contents($filePath, $content);
```
Comment (VincentLanglet) [inline src/Runner/Linter.php:82]:
```suggestion
                        file_put_contents($filePath, $newContent);
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

**Comment by rvanvelzen** [comment]
Though preventing I/O on its own is always good for performance, my main motivation was preventing auto-reloading stuff in dev when fixing did nothing.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Comment by VincentLanglet** [comment]
1.1.1 is released

**Pull request merged**